test/testdata/compiler/intrinsics/string_split.rb

```diff
@@ -0,0 +1,27 @@
+# frozen_string_literal: true
+# compiled: true
+# typed: true
+# run_filecheck: OPT
+
+module M
+  extend T::Sig
+
+  # OPT-LABEL: define internal i64 @func_M.10test_split(i32
+  sig {params(str: String).void}
+  def self.test_split(str)
+    # OPT: sorbet_rb_str_split_m
+    p str.split(' ')
+
+    # OPT: sorbet_rb_str_split_m
+    p str.split(/\s/)
+
+    # The current `String#split` intrinsic doesn't support blocks, so this case
+    # should fall back on vm dispatch
+    #
+    # OPT-NOT: sorbet_rb_str_split_m
+    p(str.split(/l/) {|x| puts "sub: #{x}"})
+  end
+  # OPT{LITERAL}: }
```

Comment:
Probably doesn't matter _too_ much for purposes of this test, but note that `split` can also take 0 arguments (the default pattern seems to be effectively `/\s+/`) or 2 (the second argument is an int limiting the number of splits, e.g. `"a:b:c:d".split(":",3)` is `["a","b","c:d"])`. Might consider throwing tests in for those.